test/python/test_mnist_training.py

```diff
@@ -38,31 +40,64 @@ class TestMnistTraining(NgraphTest):
+        # and again make sure TF and NG agree.
 
         class mnist_training_flags:
 
-            def __init__(self, data_dir, model_dir, training_iterations,
+            def __init__(self, data_dir, output_model_dir, training_iterations,
                          training_batch_size, validation_batch_size,
-                         make_deterministic, training_optimizer):
+                         make_deterministic, training_optimizer,
+                         input_model_dir):
                 self.data_dir = data_dir
-                self.model_dir = model_dir
+                self.output_model_dir = output_model_dir
                 self.train_loop_count = training_iterations
                 self.batch_size = training_batch_size
                 self.test_image_count = validation_batch_size
                 self.make_deterministic = make_deterministic
                 self.optimizer = optimizer
+                self.input_model_dir = input_model_dir
 
         data_dir = '/tmp/' + getpass.getuser() + 'tensorflow/mnist/input_data'
-        train_loop_count = 20
-        batch_size = 50
+        train_loop_count = 10
+        batch_size = 128
         test_image_count = None
         make_deterministic = True
-        model_dir = './mnist_trained/'
+        output_model_dir_ng_1 = './save_loc_ng_' + optimizer + '/'
+
+        FLAGS = mnist_training_flags(
+            data_dir, output_model_dir_ng_1, train_loop_count, batch_size,
+            test_image_count, make_deterministic, optimizer, None)
+
+        # Run on nGraph
+        ng_loss_values, ng_test_accuracy = train_mnist_cnn(FLAGS)
+        ng_values = ng_loss_values + [ng_test_accuracy]
+        # Reset the Graph
+        tf.reset_default_graph()
+
+        # disable ngraph-tf
+        ngraph_bridge.disable()
+        output_model_dir_tf_1 = './save_loc_tf_' + optimizer + '/'
+        FLAGS.output_model_dir = output_model_dir_tf_1
+        tf_loss_values, tf_test_accuracy = train_mnist_cnn(FLAGS)
+        tf_values = tf_loss_values + [tf_test_accuracy]
 
-        FLAGS = mnist_training_flags(data_dir, model_dir, train_loop_count,
-                                     batch_size, test_image_count,
-                                     make_deterministic, optimizer)
+        # compare values
+        assert np.allclose(
+            ng_values, tf_values,
+            atol=1e-3), "Loss or Accuracy values don't match"
 
+        tf.reset_default_graph()
+
+        # Now resume training from output_model_dir
+        # and dump new model in output_model_dir_ng_2
+        output_model_dir_ng_2 = './save_loc_ng_2_' + optimizer + '/'
+        FLAGS = mnist_training_flags(data_dir, output_model_dir_ng_2,
+                                     train_loop_count, batch_size,
+                                     test_image_count, make_deterministic,
+                                     optimizer, output_model_dir_ng_1)
```

Comment:
I was thinking if we can break it into two tests, but i see there is a dependency on the checkpoint being dumped